src/execution/as_abi.rs

```diff
@@ -613,3 +686,90 @@ fn alloc_string_array(env: &ASEnv, vec: &[String]) -> ABIResult<i32> {
         Err(err) => abi_bail!(err),
     }
 }
+
+#[allow(dead_code)]
+/// Tooling, return a buffer (Vec<u8>) from a given offset
+fn get_buffer(memory: &Memory, ptr: i32) -> ABIResult<Vec<u8>> {
+    match BufferPtr::new(ptr as u32).read(memory) {
+        Ok(buffer) => Ok(buffer),
+        Err(err) => abi_bail!(err),
+    }
+}
+
+/// Flatten a Vec<Vec<u8>> to a Vec<u8> with the format:
+/// L (32 bits LE) V1_L (8 bits) V1 (8bits * V1_L), V2_L ... VN (8 bits * VN_L)
+fn ser_bytearray_vec(data: &Vec<Vec<u8>>, max_entry: usize) -> ABIResult<Vec<u8>> {
+    if data.is_empty() {
+        return Ok(Vec::new());
+    }
+
+    // u16::MAX is still ok; u32::MAX -> panic!
```

Comment:
I don't understand why we set u16::MAX as the absolute maximum, shouldn't it be cleanly handled by the u32::try_from ? It would also avoid using unwrap() down there

Reply:
Alloc limit is isize::MAX which is 64 bits or 32 bits (on 32 bits platform). u16::MAX was chosen as a 'safe' but large enough upper limit and it also protects from a max_length too high (as type is usize). What do you thing?

Reply:
I think that ser_bytearray_vec should be a general-purpose function. And its max_length parameter already allows us to put a limit on the number of elements

Reply:
Removed